src/app/dialogs/Settings/Settings.jsx

```diff
@@ -0,0 +1,206 @@
+ * )
+ */
+const Settings = ({
+  hasActiveDownloads,
+  settingsDialogIsOpen
+}) => {
+  const {
+    chooseDownloadLocation,
+    setDownloadLocation,
+    setPreferenceFieldValue,
+    getPreferenceFieldValue
+  } = useContext(ElectronApiContext)
+  const [concurrentDownloads, setConcurrentDownloads] = useState('')
+  const [defaultDownloadLocation, setDefaultDownloadLocation] = useState()
+
+  const onClearDefaultDownload = () => {
+    setDefaultDownloadLocation(undefined)
+    setPreferenceFieldValue('defaultDownloadLocation', undefined)
+  }
+
+  const onSetChooseDownloadLocation = () => {
+    chooseDownloadLocation()
+  }
+
+  const onChangeConcurrentDownloads = (event) => {
+    const { value } = event.target
+    // if value is non numerical and can't be parsed as an integer return 0
+    const valueNumeric = parseInt(value, 10) || 0
+    if (value === '') {
+      setConcurrentDownloads('')
+      return
+    }
+    // Allow set if non-decimal value and > 0 allow setting of the text field
+    if (valueNumeric > 0 && value.indexOf('.') < 0) {
+      setConcurrentDownloads(valueNumeric.toString())
+    }
+  }
+  // Event occurs when element loses focus, write to preferences.json
+  const onBlurConcurrentDownloads = async (event) => {
+    const { value } = event.target
+    // If empty string is entered and the input field is exited, restore the label with the set concurrentDownloads
+    if (!value) {
+      const concurrentDownloadsPreference = await getPreferenceFieldValue('concurrentDownloads')
+      setConcurrentDownloads(concurrentDownloadsPreference.toString())
+      return
+    }
+
+    const valueNumeric = parseInt(value, 10)
+    if (!Number.isNaN(valueNumeric) && valueNumeric > 0) {
+      setPreferenceFieldValue('concurrentDownloads', valueNumeric)
+    }
+  }
+
+  const onSetDownloadLocation = (event, info) => {
+    const { downloadLocation: newDownloadLocation } = info
+    setDefaultDownloadLocation(newDownloadLocation)
+    setPreferenceFieldValue('defaultDownloadLocation', newDownloadLocation)
+  }
+
+  useEffect(
+    () => {
+      const fetchDefaultDownloadLocation = async () => {
+        setDefaultDownloadLocation(await getPreferenceFieldValue('defaultDownloadLocation'))
+      }
+      fetchDefaultDownloadLocation()
+    },
+    [setDefaultDownloadLocation]
+  )
+
+  useEffect(() => {
+    const fetchConcurrency = async () => {
+      const concurrentDownloads = await getPreferenceFieldValue('concurrentDownloads')
+      setConcurrentDownloads(concurrentDownloads.toString())
+    }
+    fetchConcurrency()
+  }, [setConcurrentDownloads])
+
+  // Handle the response from the setDownloadLocation
+  useEffect(() => {
+    setDownloadLocation(true, onSetDownloadLocation)
+    return () => {
+      setDownloadLocation(false, onSetDownloadLocation)
+    }
+  }, [])
+
+  useEffect(() => {
+    // Handle edge case where change is made to the concurrency field but, exits
+    if (settingsDialogIsOpen === false) {
```

Comment:
```suggestion
    if (!settingsDialogIsOpen) {
```
What about if other fields change as well but exit without applying the changes?

Reply:
Yea that can't really happen because the other are buttons and a folder selector. The concurrency is the only "Text field"